src/app/api/query/notifications.ts

```diff
@@ -0,0 +1,117 @@
+import { useEffect } from "react";
+
+import { useToastNotification } from "@canonical/react-components";
+import {
+  useMutation,
+  useQueryClient,
+  type UseQueryOptions,
+} from "@tanstack/react-query";
+
+import { useWebsocketAwareQuery } from "./base";
+
+import type {
+  DismissNotificationError,
+  DismissNotificationResponse,
+  DismissNotificationData,
+  ListNotificationsData,
+  ListNotificationsError,
+  ListNotificationsResponse,
+  Options,
+} from "@/app/apiclient";
+import {
+  dismissNotificationMutation,
+  listNotificationsOptions,
+  listNotificationsQueryKey,
+} from "@/app/apiclient/@tanstack/react-query.gen";
+
+export const useListNotifications = (
+  options?: Options<ListNotificationsData>
+) => {
+  return useWebsocketAwareQuery(
+    listNotificationsOptions(options) as UseQueryOptions<
+      ListNotificationsData,
+      ListNotificationsError,
+      ListNotificationsResponse
+    >
+  );
+};
+
+export const useNotifications = () => {
+  const backendNotifications = useListNotifications();
+  const items = backendNotifications.data?.items;
+  const notifications = useToastNotification();
+  const dismissMutatation = useDismissNotification();
+  useEffect(() => {
+    if (items === undefined) return;
+    items.forEach((item) => {
+      switch (item.category) {
+        case "success":
+          notifications.success(item.message, [
+            {
+              label: "Dismiss",
+              onClick: () => {
+                dismissMutatation.mutate({
+                  path: { notification_id: item.id },
+                });
+              },
+            },
+          ]);
```

Comment:
You can give an empty string as the last argument here to get rid of the "Success" titles - IMO they look a bit awkward, but up to you
```suggestion
          ], "");
```